pkg/manifests/config.go

```diff
@@ -197,6 +208,24 @@ func NewConfig(content io.Reader) (*Config, error) {
 	res := &c
 	res.applyDefaults()
 	c.UserWorkloadConfiguration = NewDefaultUserWorkloadMonitoringConfig()
+	// The operator should only create some manifests if techPreview is enabled
+	c.TechPreview = tp
+
+	if c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile != FullCollectionProfile && !tp {
+		return nil, errors.Wrap(ErrConfigValidation, "collectionProfiles is a TechPreview feature, to be able to use a profile different from the default (\"full\") please enable TechPreview")
+	}
+	// Validate CollectionProfile field
+	foundProfile := false
+	for _, profile := range SupportedCollectionProfiles {
+		if profile == c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile {
+			foundProfile = true
+			continue
+		}
+	}
+	if !foundProfile {
```

Comment:
may be replaced with `slices.Contains(SupportedCollectionProfiles, profile)` 
see: https://pkg.go.dev/golang.org/x/exp/slices#Contains